File: LivrusVirtus/wxWindows/wxRuby/Filha.rb
require 'wxruby'
require 'Principal'
require 'CadAssuntos'
require 'CadAutores'
require 'CadEditoras'
require 'CadEnderecos'
require 'CadIdiomas'
require 'CadPaises'
require 'CadProfissoes'
require 'CadClientes'
require 'CadLivros'
require 'VenderLivros'
require 'ConVendas'
require 'AlterarSenha'

class FrmFilha < Wx::MDIChildFrame
  def initialize(parent, nTelaAtiva)
    super(parent, -1, "Cadastro de ", Wx::Point.new(0, 0), 
      Wx::Size.new(342, 219), Wx::DEFAULT_FRAME_STYLE,"") 
    set_client_size(Wx::Size.new(330, 136))
    evt_close {|event| on_close(event)}

    if Wx::RUBY_PLATFORM == "WXMSW"
      set_icon(Wx::Icon.new("xpms\\Livrus.ico"))
    else
      set_icon(Wx::Icon.new("xpms\\livrus.xpm"))
    end
    @nLocalTelaAtiva = nTelaAtiva

    if nTelaAtiva == Resource::FRMCADASSUNTOS then
      FrmCadAssuntos.new(self)
      set_title(Resource::STR_CADASSUNTO)
    elsif nTelaAtiva == Resource::FRMCADAUTORES then
      FrmCadAutores.new(self)
      set_title(Resource::STR_CADAUTOR)
    elsif nTelaAtiva == Resource::FRMCADEDITORAS then
      FrmCadEditoras.new(self)
      set_title(Resource::STR_CADEDITORA)
    elsif nTelaAtiva == Resource::FRMCADENDERECOS then
      FrmCadEnderecos.new(self)
      set_title(Resource::STR_CADENDERECO)
      set_client_size(Wx::Size.new(525, 170))
    elsif nTelaAtiva == Resource::FRMCADIDIOMAS then
      FrmCadIdiomas.new(self)
      set_title(Resource::STR_CADIDIOMA)        
    elsif nTelaAtiva == Resource::FRMCADPAISES then
      FrmCadPaises.new(self)
      set_title(Resource::STR_CADPAIS)
    elsif nTelaAtiva == Resource::FRMCADPROFISSOES then
      FrmCadProfissoes.new(self)
      set_title(Resource::STR_CADPROFISS)
    elsif nTelaAtiva == Resource::FRMCADCLIENTES then
      $fCadClientes = FrmCadClientes.new(self)
      set_title(Resource::STR_CADCLIENTE)
      set_client_size(Wx::Size.new(525, 440))
    elsif nTelaAtiva == Resource::FRMCADLIVROS then
      $fCadLivros = FrmCadLivros.new(self)
      set_title(Resource::STR_CADLIVRO)
      set_client_size(Wx::Size.new(510, 425))        
    elsif nTelaAtiva == Resource::FRMVENDERLIVROS then
      $fVenderLivros = FrmVenderLivros.new(self)
      set_title(Resource::STR_VENLIVRO)
      set_client_size(Wx::Size.new(480, 310))        
    elsif nTelaAtiva == Resource::FRMCONVENDAS then
      $fConVendas = FrmConVendas.new(self)
      set_title(Resource::STR_CONVENLIVRO)
      set_client_size(Wx::Size.new(480, 330))
    elsif nTelaAtiva == Resource::FRMALTERARSENHA then
      FrmAlterarSenha.new(self)
      set_title(Resource::STR_ALTSENHA)
      set_client_size(Wx::Size.new(255, 134))
    end
  end

  def on_close(event)
    if @nLocalTelaAtiva == Resource::FRMCADASSUNTOS then
      $fPrincipal.retMenu(0).enable(Resource::MNU_CAD_ASSUNTOS, true) 
    elsif @nLocalTelaAtiva == Resource::FRMCADAUTORES then 
      $fPrincipal.retMenu(0).enable(Resource::MNU_CAD_AUTORES, true)  
    elsif @nLocalTelaAtiva == Resource::FRMCADEDITORAS then
      $fPrincipal.retMenu(0).enable(Resource::MNU_CAD_EDITORAS, true)    
    elsif @nLocalTelaAtiva == Resource::FRMCADENDERECOS then
      $fPrincipal.retMenu(0).enable(Resource::MNU_CAD_ENDERECOS, true)    
    elsif @nLocalTelaAtiva == Resource::FRMCADIDIOMAS then
      $fPrincipal.retMenu(0).enable(Resource::MNU_CAD_IDIOMAS, true)   
    elsif @nLocalTelaAtiva == Resource::FRMCADPAISES then 
      $fPrincipal.retMenu(0).enable(Resource::MNU_CAD_PAISES, true)    
    elsif @nLocalTelaAtiva == Resource::FRMCADPROFISSOES then 
      $fPrincipal.retMenu(0).enable(Resource::MNU_CAD_PROFISSOES, true)    
    elsif @nLocalTelaAtiva == Resource::FRMCADCLIENTES then
      $fPrincipal.retBarraFerra().enable_tool(Resource::MNU_CAD_CLIENTES, true)
      $fPrincipal.retMenu(0).enable(Resource::MNU_CAD_CLIENTES, true)   
    elsif @nLocalTelaAtiva == Resource::FRMCADLIVROS then
      $fPrincipal.retBarraFerra().enable_tool(Resource::MNU_CAD_LIVROS, true)
      $fPrincipal.retMenu(0).enable(Resource::MNU_CAD_LIVROS, true)
    elsif @nLocalTelaAtiva == Resource::FRMVENDERLIVROS then
      $fPrincipal.retBarraFerra().enable_tool(Resource::MNU_VEN_VENDERLIVROS, 
        true)
      $fPrincipal.retMenu(2).enable(Resource::MNU_VEN_VENDERLIVROS, true)  
    elsif @nLocalTelaAtiva == Resource::FRMCONVENDAS then 
      $fPrincipal.retMenu(2).enable(Resource::MNU_VEN_VENDASREALIZADAS, true)
    elsif @nLocalTelaAtiva == Resource::FRMALTERARSENHA then
      $fPrincipal.retMenu(3).enable(Resource::MNU_OP_ALTERARSENHA, true)
    end
    event.skip()
  end

end

File: LivrusVirtus/BoaConstructor/wxPython/Filha.py
#Boa:MDIChild:frmFilha

from wxPython.wx import *
from resource import Resource
import CadAssuntos
import CadAutores
import CadEditoras
import CadEnderecos
import CadIdiomas
import CadPaises
import CadProfissoes
import CadClientes
import CadLivros
import VenderLivros
import ConVendas
import AlterarSenha
import Principal

def create(parent): 
    return frmFilha(parent)

[wxID_FRMFILHA] = map(lambda _init_ctrls: wxNewId(), range(1))

class frmFilha(wxMDIChildFrame):
    def _init_utils(self):
        # generated method, don't edit
        pass

    def _init_ctrls(self, prnt):
        # generated method, don't edit
        wxMDIChildFrame.__init__(self, id=wxID_FRMFILHA, name=u'frmFilha',
              parent=prnt, pos=wxPoint(0, 0), size=wxSize(342, 219),
              style=wxDEFAULT_FRAME_STYLE, title=u'Cadastro de ')
        self._init_utils()
        self.SetIcon(wxIcon(Resource.ICO_LIVRUS, wxBITMAP_TYPE_ICO))
        self.SetClientSize(wxSize(330, 136))
        EVT_CLOSE(self, self.OnFrmFilhaClose)

        if prnt.nTelaAtiva == Resource.FRMCADASSUNTOS:
          CadAssuntos.create(self)
          self.SetTitle(Resource.STR_CADASSUNTO)
        elif prnt.nTelaAtiva == Resource.FRMCADAUTORES:
          CadAutores.create(self)
          self.SetTitle(Resource.STR_CADAUTOR)
        elif prnt.nTelaAtiva == Resource.FRMCADEDITORAS:
          CadEditoras.create(self)
          self.SetTitle(Resource.STR_CADEDITORA)
        elif prnt.nTelaAtiva == Resource.FRMCADENDERECOS:
          CadEnderecos.create(self)
          self.SetTitle(Resource.STR_CADENDERECO)
          self.SetClientSize(wxSize(525, 170))
        elif prnt.nTelaAtiva == Resource.FRMCADIDIOMAS:
          CadIdiomas.create(self)
          self.SetTitle(Resource.STR_CADIDIOMA)        
        elif prnt.nTelaAtiva == Resource.FRMCADPAISES:
          CadPaises.create(self)
          self.SetTitle(Resource.STR_CADPAIS)
        elif prnt.nTelaAtiva == Resource.FRMCADPROFISSOES:
          CadProfissoes.create(self)
          self.SetTitle(Resource.STR_CADPROFISS)
        elif prnt.nTelaAtiva == Resource.FRMCADCLIENTES:
          CadClientes.create(self)
          self.SetTitle(Resource.STR_CADCLIENTE)
          self.SetClientSize(wxSize(525, 440))
        elif prnt.nTelaAtiva == Resource.FRMCADLIVROS:
          CadLivros.create(self)
          self.SetTitle(Resource.STR_CADLIVRO)
          self.SetClientSize(wxSize(510, 425))        
        elif prnt.nTelaAtiva == Resource.FRMVENDERLIVROS:
          VenderLivros.create(self)
          self.SetTitle(Resource.STR_VENLIVRO)
          self.SetClientSize(wxSize(480, 310))        
        elif prnt.nTelaAtiva == Resource.FRMCONVENDAS:
          ConVendas.create(self)
          self.SetTitle(Resource.STR_CONVENLIVRO)
          self.SetClientSize(wxSize(480, 330))
        elif prnt.nTelaAtiva == Resource.FRMALTERARSENHA:
          AlterarSenha.create(self)
          self.SetTitle(Resource.STR_ALTSENHA)
          self.SetClientSize(wxSize(255, 134))

    def __init__(self, parent):
        self._init_ctrls(parent)

    def OnFrmFilhaClose(self, event):
      frmPrincipal = Principal.fPrincipal
      if self.GetParent().nTelaAtiva == Resource.FRMCADASSUNTOS:
        frmPrincipal.BarraDeMenu.GetMenu(0).Enable(Resource.MNU_CAD_ASSUNTOS, True)
      elif self.GetParent().nTelaAtiva == Resource.FRMCADAUTORES: 
        frmPrincipal.BarraDeMenu.GetMenu(0).Enable(Resource.MNU_CAD_AUTORES, True)  
      elif self.GetParent().nTelaAtiva == Resource.FRMCADEDITORAS:
        frmPrincipal.BarraDeMenu.GetMenu(0).Enable(Resource.MNU_CAD_EDITORAS, True)    
      elif self.GetParent().nTelaAtiva == Resource.FRMCADENDERECOS:
        frmPrincipal.BarraDeMenu.GetMenu(0).Enable(Resource.MNU_CAD_ENDERECOS, True)    
      elif self.GetParent().nTelaAtiva == Resource.FRMCADIDIOMAS:
        frmPrincipal.BarraDeMenu.GetMenu(0).Enable(Resource.MNU_CAD_IDIOMAS, True)   
      elif self.GetParent().nTelaAtiva == Resource.FRMCADPAISES: 
        frmPrincipal.BarraDeMenu.GetMenu(0).Enable(Resource.MNU_CAD_PAISES, True)    
      elif self.GetParent().nTelaAtiva == Resource.FRMCADPROFISSOES: 
        frmPrincipal.BarraDeMenu.GetMenu(0).Enable(Resource.MNU_CAD_PROFISSOES, True)    
      elif self.GetParent().nTelaAtiva == Resource.FRMCADCLIENTES:
        frmPrincipal.tlbCadClientes.Enable(True)
        frmPrincipal.BarraDeMenu.GetMenu(0).Enable(Resource.MNU_CAD_CLIENTES, True)   
      elif self.GetParent().nTelaAtiva == Resource.FRMCADLIVROS:
        frmPrincipal.tlbCadLivros.Enable(True)
        frmPrincipal.BarraDeMenu.GetMenu(0).Enable(Resource.MNU_CAD_LIVROS, True)
      elif self.GetParent().nTelaAtiva == Resource.FRMVENDERLIVROS:
        frmPrincipal.tlbVenderLivros.Enable(True)
        frmPrincipal.BarraDeMenu.GetMenu(2).Enable(Resource.MNU_VEN_VENDERLIVROS, True)  
      elif self.GetParent().nTelaAtiva == Resource.FRMCONVENDAS: 
        frmPrincipal.BarraDeMenu.GetMenu(2).Enable(Resource.MNU_VEN_VENDASREALIZADAS, True)
      elif self.GetParent().nTelaAtiva == Resource.FRMALTERARSENHA:
        frmPrincipal.BarraDeMenu.GetMenu(3).Enable(Resource.MNU_OP_ALTERARSENHA, True)
      event.Skip()
    

